scripts/multicluster/setup.sh

```diff
@@ -0,0 +1,176 @@
+done
+
+# ==== Deploy a curl client for testing
+${k1} apply -f - <<EOF
+apiVersion: v1
+kind: Deployment
+metadata:
+  name: curl
+  namespace: bookbuyer
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: curl
+  template:
+    metadata:
+      labels:
+        app: curl
+    spec:
+      containers:
+      - name: curl
+        image: curlimages/curl
+        command: ["sleep", "1000000"]
+EOF
+
+# ==== 3. Create ServiceExport
+${k2} apply -f - <<EOF
+apiVersion: multicluster.x-k8s.io/v1alpha1
+kind: ServiceExport
+metadata:
+  name: bookstore
+  namespace: bookstore
+EOF
+
+# ==== 4. Create ServiceImport
+
+# create a meta-service to get a valid local service IP
+${k1} apply -f - <<EOF
+apiVersion: v1
+kind: Service
+metadata:
+  name: mcs-bookstore
+  namespace: bookstore
+spec:
+  type: ClusterIP
+  ports:
+  - port: ${app_port}
+    protocol: TCP
+EOF
+sleep 5
+mcs_ip=$(${k1} get service mcs-bookstore -n bookstore -o jsonpath='{.spec.clusterIP}')
+
+${k1} apply -f - <<EOF
+apiVersion: multicluster.x-k8s.io/v1alpha1
+kind: ServiceImport
+metadata:
+  name: bookstore
+  namespace: bookstore
+spec:
+  ips:
+  - ${mcs_ip}
+  type: ClusterSetIP
+  ports:
+  - name: http
+    protocol: TCP
+    port: ${app_port}
+EOF
+
+# ==== 5. Expose service on c2 and get the IP addresses.
+${k2} patch service bookstore -n bookstore -p '{"spec":{"type":"LoadBalancer"}}'
+for _ in {1..10}; do
+    remote_ip=$(${k2} get service bookstore -n bookstore -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+
+    if [ -n "${remote_ip}" ]; then
+        break
+    fi
+    sleep 5
+done
+if [ -z "${remote_ip}" ]; then
+    echo "Remote bookstore IP not found"
+    exit 1
+fi
+echo "Remote bookstore IP: ${remote_ip}"
+
+local_ip=$(${k1} get service bookstore -n bookstore -o jsonpath='{.spec.clusterIP}')
+
+# ==== 6. Create remote endpointSlice
+uid=$(${k1} get serviceimport bookstore -n bookstore -o jsonpath='{.metadata.uid}')
+${k1} apply -f - <<EOF
+apiVersion: discovery.k8s.io/v1
+kind: EndpointSlice
+metadata:
+  name: imported-bookstore-cluster-c2
+  namespace: bookstore
+  labels:
+    multicluster.kubernetes.io/source-cluster: cluster-c2
+    multicluster.kubernetes.io/service-name: bookstore
+  ownerReferences:
+  - apiVersion: multicluster.k8s.io/v1alpha1
+    controller: false
+    kind: ServiceImport
+    name: bookstore
+    uid: ${uid}
+addressType: IPv4
+ports:
+  - name: http
+    protocol: TCP
+    port: ${app_port}
+endpoints:
+  - addresses:
+    - "${remote_ip}"
+EOF
+
+# ==== 7. Update coreDNS entries
+${k1} apply -f - <<EOF
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: coredns-custom
+  namespace: kube-system
+data:
+  mcs-bookstore.server: |
+    bookstore.bookstore.svc.clusterset.local:53 {
+      errors
+      cache 30
+      hosts {
+        ${mcs_ip} bookstore.bookstore.svc.clusterset.local
+        ${local_ip} bookstore.bookstore.svc.clusterset.local
```

Comment:
This should not resolve to the local ip, only the above mcs_ip is necessary.

Remember it is kube proxy that knows how to resolve from the ClusterSetIP to the appropriate set of EndpointSlices

Reply:
Kube-proxy doesn't natively know how to resolve to clusterset.local  without an update right?

Reply:
Correct, but our traffic also doesn't go through kube-proxy, so we'll need to bring the functionality to the sidecar either way

Reply:
Ah yes, I was just addressing this:
> Remember it is kube proxy that knows how to resolve from the ClusterSetIP to the appropriate set of EndpointSlices

Did you perhaps mean that kube-proxy knows how to resolve the local ClusterIP rather than the ClustersetIP?